File: Abstract-Factory/Python/dark_theme_factory.py
from ui_factory import UIFactory
from dark_button import DarkButton
from dark_checkbox import DarkCheckbox

class DarkThemeFactory(UIFactory):
    def create_button(self) -> DarkButton:
        return DarkButton()

    def create_checkbox(self) -> DarkCheckbox:
        return DarkCheckbox()

File: Abstract-Factory/PHP/DarkThemeFactory.php
<?php

declare(strict_types=1);

require_once __DIR__ . '/UIFactory.php';
require_once __DIR__ . '/DarkButton.php';
require_once __DIR__ . '/DarkCheckbox.php';

class DarkThemeFactory implements UIFactory {
    public function createButton(): Button {
        return new DarkButton();
    }

    public function createCheckbox(): Checkbox {
        return new DarkCheckbox();
    }
}
